dateparser/date.py

```diff
@@ -175,7 +175,7 @@ def _try_given_formats(self):
         if not self.date_formats:
             return
 
-        return parse_with_formats(self._get_translated_date_with_formatting(), self.date_formats)
+        return parse_with_formats(self._get_translated_date_with_formatting(), list(self.date_formats))
```

Comment:
I could get behind that check, works for me. When I get back on my laptop
I'll make the change and run the tests

On Mon, Feb 9, 2015, 5:44 PM Eugene Amirov <notifications@github.com> wrote:

> In dateparser/date.py
> <https://github.com/scrapinghub/dateparser/pull/18#discussion_r24373150>:
>
> > @@ -175,7 +175,7 @@ def _try_given_formats(self):
> >          if not self.date_formats:
> >              return
> >
> > -        return parse_with_formats(self._get_translated_date_with_formatting(), self.date_formats)
> > +        return parse_with_formats(self._get_translated_date_with_formatting(), list(self.date_formats))
>
> We used this check
> <https://github.com/scrapinghub/dateparser/blob/master/dateparser/date.py#L222>
> for similar case. May be it is worth to add this group to dateparser.utils
> as plain_iterable or something. What do you think?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scrapinghub/dateparser/pull/18/files#r24373150>.
>